src/gen.rs

```diff
@@ -430,6 +430,10 @@ fn gen_method_item(
     // Generate the list of argument used to call the method.
     let args = get_arg_list(sig.decl.inputs.iter())?;
 
+    // Determines the number of generic parameters, and what they are
+    let type_param_count = sig.decl.generics.type_params().count();
+    let (_, generic_types, _ ) = &sig.decl.generics.split_for_impl();
```

Comment:
The space between `_` and `)` should be removed.

Why is there a `&` before `sig`? I don't think that's necessary, so we could remove it.